cmake/AddOsqpEigenUnitTest.cmake

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+#
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+
+find_package(Catch2 QUIET)
+checkandset_dependency(Catch2)
+
+find_package(VALGRIND QUIET)
+checkandset_dependency(VALGRIND)
+
+osqpeigen_dependent_option(OSQPEIGEN_COMPILE_tests
+  "Compile tests?" ON
+  "OSQPEIGEN_HAS_Catch2;BUILD_TESTING" OFF)
+
+osqpeigen_dependent_option(OSQPEIGEN_RUN_Valgrind_tests
+  "Run Valgrind tests?" OFF
+  "OSQPEIGEN_COMPILE_tests;VALGRIND_FOUND" OFF)
+
+if (OSQPEIGEN_RUN_Valgrind_tests)
+    set(CTEST_MEMORYCHECK_COMMAND ${VALGRIND_PROGRAM})
+    set(MEMORYCHECK_COMMAND ${VALGRIND_PROGRAM})
+    if (APPLE)
+        set(MEMORYCHECK_SUPPRESSIONS "--suppressions=${PROJECT_SOURCE_DIR}/cmake/valgrind-macos.supp")
+    else ()
+        set(MEMORYCHECK_SUPPRESSIONS "")
+    endif ()
+    set(MEMORYCHECK_COMMAND_OPTIONS "--leak-check=full --error-exitcode=1 ${MEMORYCHECK_SUPPRESSIONS}"  CACHE STRING "Options to pass to the memory checker")
+    mark_as_advanced(MEMORYCHECK_COMMAND_OPTIONS)
+    set(MEMCHECK_COMMAND_COMPLETE "${MEMORYCHECK_COMMAND} ${MEMORYCHECK_COMMAND_OPTIONS}")
+    separate_arguments(MEMCHECK_COMMAND_COMPLETE)
+endif()
+
+if (OSQPEIGEN_COMPILE_tests)
+    configure_file(cmake/Catch2Main.cpp.in ${CMAKE_BINARY_DIR}/Testing/Catch2Main.cpp)
+    add_library(CatchTestMain ${CMAKE_BINARY_DIR}/Testing/Catch2Main.cpp)
+    target_link_libraries(CatchTestMain PUBLIC Catch2::Catch2)
+endif()
+
+
+function(add_osqpeigen_test)
+
+    if(OSQPEIGEN_COMPILE_tests)
+
+      set(options DEPENDS_ON_EIGEN_PRIVATE)
```

Comment:
Is this necessary?

Reply:
I don't think so

Reply:
Removed in 24b77cac3a19e43ae2bbf663b2ee4a450d5130ca